src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -209,12 +245,6 @@ int main(int argc, char *argv[])
     Log::Info << "Loaded kNN model from '" << inputModelFile << "' (trained on "
         << knn.Dataset().n_rows << "x" << knn.Dataset().n_cols << " dataset)."
         << endl;
-
-    // Adjust singleMode and naive if necessary.
-    knn.SingleMode() = CLI::HasParam("single_mode");
-    knn.Naive() = CLI::HasParam("naive");
-    knn.LeafSize() = size_t(lsInt);
-    knn.Epsilon() = epsilon;
```

Comment:
Also, another comment, I think we should provide a command line option:
```
  --mode (-m) [string]
```
So, we can specify different search modes:  *naive*,  *single_mode*, *dual_mode*, *greedy*

Then, when an input model file is provided, we could check if the option "--mode" was specified. In that case, we update the input model file to the given mode.